tools/db/cleanUpWhisks.py

```diff
@@ -95,56 +98,108 @@ def deleteWhisk(dbWhisks, wdoc):
 def checkNamespace(dbSubjects, namespace):
 
     while True:
-
         allNamespaces = dbSubjects.view('subjects/identities', startkey=[namespace], endkey=[namespace])
-
         if allNamespaces:
             return True
         else:
             return False
 
 
+#
+# update last namespace info and print number of kept documents
+#
+def updateLastNamespaceInfo(namespace, lastNamespaceInfo):
+
+    if namespace != lastNamespaceInfo[0]:
+        if lastNamespaceInfo[0] != None:
+            print('keeping: {0} doc(s) for namespace {1}'.format(lastNamespaceInfo[1], lastNamespaceInfo[0]))
+        lastNamespaceInfo = (namespace,0)
+    lastNamespaceInfo = (namespace, lastNamespaceInfo[1]+1)
+    return lastNamespaceInfo
+
+
+#
+# get delimiters for namespace retrieval by database type
+#
+def getDelimiters(whiskDatabaseType):
+
+    if whiskDatabaseType == "whisks":
+        # "_id": "whisk.system/utils",
+        return None, '/'
+    elif whiskDatabaseType == "cloudanttrigger":
+        # "_id": ":AHA04676_dev:vision-cloudant-trigger",
```

Comment:
this seems like over specialization - The script should be generic and all details like the database type below passed in as parameters.

Reply:
@rabbah didn't get your last point, the database type is passed as parameter (`--whiskDBType`) and by that the delimiters for the namespace extract from the document id are determined.

Reply:
Right, thanks for following up. I am asking if the delimiter can be generic parameter (or the parts relayed differently) as opposed to teaching this script about the various databases for providers which are orthogonal to the openwhisk backend.

Reply:
@rabbah ah ok, @cbickel and me discussed that point and thought it would be better to hide the internals and let the caller only specify the type of the whisk database that needs to be cleaned up, otherwise you would require knowledge about how the documents look like and what the script will done in order to obtain the namespace to check. I will discuss with @cbickel tomorrow about your point.

Reply:
The purpose of this script is, that an operator is able to cleanup the whisks-databases of his installation he is managing. He necessarily doesn't know, how the id of the documents is structured for each database.
So I think it would be valid to just pass the database name and the database type to hide these internals.
Do you agree, Rodric?